libs/feature/dataviz/src/lib/geo-table-view/geo-table-view.component.ts

```diff
@@ -1,40 +1,168 @@
-import { Component } from '@angular/core'
 import {
-  FEATURE_COLLECTION_POINT_FIXTURE_4326,
+  AfterViewInit,
+  ChangeDetectionStrategy,
+  ChangeDetectorRef,
+  Component,
+  Input,
+  OnDestroy,
+  OnInit,
+  ViewChild,
+} from '@angular/core'
+import {
+  FeatureInfoService,
   MAP_CTX_LAYER_XYZ_FIXTURE,
   MapContextLayerTypeEnum,
-} from '@geonetwork-ui/ui/map'
-import { fromLonLat } from 'ol/proj'
+  MapContextModel,
+  MapManagerService,
+} from '@geonetwork-ui/feature/map'
+import { TableComponent } from '@geonetwork-ui/ui/layout'
+import { FEATURE_COLLECTION_POINT_FIXTURE_4326 } from '@geonetwork-ui/util/shared'
+import { FeatureCollection } from 'geojson'
+import { Feature, Map, View } from 'ol'
+import { Geometry } from 'ol/geom'
+import VectorLayer from 'ol/layer/Vector'
+import VectorSource from 'ol/source/Vector'
+import { Circle, Fill, Stroke } from 'ol/style'
+import Style from 'ol/style/Style'
+import { Subscription } from 'rxjs'
+
+const defaultStyle: Style = new Style({
+  image: new Circle({
+    fill: new Fill({
+      color: 'blue',
+    }),
+    stroke: new Stroke({
+      color: 'blue',
+      width: 1,
+    }),
+    radius: 5,
+  }),
+})
+
+const HLStyle: Style = new Style({
+  image: new Circle({
+    fill: new Fill({
+      color: 'red',
+    }),
+    stroke: new Stroke({
+      color: 'red',
+      width: 2,
+    }),
+    radius: 5,
+  }),
+})
 
 @Component({
   selector: 'gn-ui-geo-table-view',
   templateUrl: './geo-table-view.component.html',
   styleUrls: ['./geo-table-view.component.css'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
 })
-export class GeoTableViewComponent {
-  geojson = FEATURE_COLLECTION_POINT_FIXTURE_4326
+export class GeoTableViewComponent implements OnInit, AfterViewInit, OnDestroy {
+  @Input() data: FeatureCollection = FEATURE_COLLECTION_POINT_FIXTURE_4326
+  @ViewChild(TableComponent) uiTable: TableComponent
+
+  private map: Map
+  private view: View
+  private vectorLayer: VectorLayer<VectorSource<Geometry>>
+  private vectorSource: VectorSource<Geometry>
+  private features: Feature<Geometry>[]
+
+  tableData: any
+  mapContext: MapContextModel
+  selectionId: any
+  selection: Feature<Geometry>
+  private subscription = new Subscription()
 
-  get tableData() {
-    return this.geojson.features.map((f) => ({
+  constructor(
+    private manager: MapManagerService,
+    private featureInfo: FeatureInfoService,
+    private changeRef: ChangeDetectorRef
+  ) {}
+
+  ngOnInit(): void {
+    this.tableData = this.geojsonToTableData(this.data)
```

Comment:
The component should advertise that it will not update if the input data changes IMO (or a `ngChanges` handler should be added, but that might be non trivial)